### PR TITLE
Feat: change request scheduled timeline

### PR DESCRIPTION
**Pull request by andreas-unleash**
Adds the scheduled state to the timeline

Closes # [1-1632](https://linear.app/unleash/issue/1-1632/update-the-progress-bar-for-scheduled-changes)

Screenshot 2023-11-16 at 10 51 03

### CODE REVIEW

File: frontend/src/component/changeRequest/ChangeRequestOverview/ChangeRequestTimeline/ChangeRequestTimeline.tsx [hunk 1]
```diff
@@ -44,28 +52,53 @@ export const determineColor = (
     displayStageIndex: number,
 ) => {
     if (changeRequestState === 'Cancelled') return 'grey';
+
     if (changeRequestState === 'Rejected')
         return displayStage === 'Rejected' ? 'error' : 'success';
     if (
         changeRequestStateIndex !== -1 &&
-        changeRequestStateIndex >= displayStageIndex
```
Comment (FredrikOseberg) [inline frontend/src/component/changeRequest/ChangeRequestOverview/ChangeRequestTimeline/ChangeRequestTimeline.tsx:51]: What did this solve?

File: frontend/src/component/changeRequest/ChangeRequestOverview/ChangeRequestTimeline/ChangeRequestTimeline.tsx [hunk 2]
```diff
@@ -106,6 +141,14 @@ const createTimelineItem = (
...
         </TimelineSeparator>
-        <TimelineContent>{title}</TimelineContent>
+        <TimelineContent>
+            {title}
+            <br />
+            {subtitle && (
+                <Typography
+                    color={'text.secondary'}
+                >{`(for ${subtitle})`}</Typography>
+            )}
```
Comment (FredrikOseberg) [inline frontend/src/component/changeRequest/ChangeRequestOverview/ChangeRequestTimeline/ChangeRequestTimeline.tsx:151]: Until we change the ADR we should use ConditionallyRender for consistency

**Review by FredrikOseberg — APPROVED** [review]
LGTM, minor comments

File: frontend/src/component/changeRequest/ChangeRequestOverview/ChangeRequestTimeline/ChangeRequestTimeline.tsx (hunk 1 above)
Comment (andreas-unleash) [reply frontend/src/component/changeRequest/ChangeRequestOverview/ChangeRequestTimeline/ChangeRequestTimeline.tsx:51]: We want a `warning` color when change request is in `Scheduled` state where as all the other states have a `success` color.  This is why the `=` was handled separately

**Pull request merged**